sdk/core/azure-core/src/datetime.cpp

```diff
@@ -729,42 +729,39 @@ DateTime DateTime::Parse(std::string const& dateTime, DateFormat format)
   }
 }
 
-std::string DateTime::ToString(DateFormat format, TimeFractionFormat fractionFormat) const
+void DateTime::ValidateYear() const
```

Comment:
For Context we use `ThrowIfCancelled()` ,  should we do the same here and change the name to `ThrowIfInvalidDate()` ??
and have one `bool IsValid()`

Reply:
Those are public APIs, this is not and is just an implementation detail. That said, I do like your suggestion, how about: `ThrowIfUnsupportedYear()`